File: api/Models/Review.js
const db = require("../../db/connect")

class Review {
    constructor({id, rating, movie_name, user_id}){
        this.id = id,
        this.rating = rating,
        this.movie_name = movie_name,
        this.user_id = user_id
    }

    static async getAllByUserID (user_id) {
        const response = await db.query("SELECT * FROM review WHERE user_id = $1:", [user_id])

        if(response.rows.length === 0){
            throw new Error("No reviews from this user")
        }
        return( response.rows.map(r => new Review(r)))
    }

    static async createReview (reviewData) {
        if(!userData.movie_name){throw new Error("Movie name is missing")}
        if(!userData.user_id){throw new Error("User ID is missing")}

        const response = await db.query("INSERT INTO review (rating, movie_name, user_id) VALUES($1, $2, $3) RETURNING *:", [reviewData.rating, reviewData.movie_name, reviewData.user_id])
        return new Review(response)
    }
}

module.exports = Review

File: python/models/Review.py
from models.db.connect import cur, conn

class Review:

    def __init__(self, id, rating, song_name, artist):
        self.id = id
        self.rating = rating
        self.song_name = song_name
        self.artist = artist

    def to_dict(self):
        return {"id": self.id, "rating":self.rating, "song_name": self.song_name, "artist": self.artist}

    def get_all():
        cur.execute("SELECT * FROM review")
        revs = cur.fetchall()
        reviews = [Review(id=rev[0], rating=rev[1], song_name=rev[2], artist=rev[3]) for rev in revs]
        conn.commit()
        return reviews

    def create_review(rating, song_name, artist):
        cur.execute("INSERT INTO review (rating, song_name, artist) VALUES (%s, %s, %s) RETURNING *", [rating, song_name, artist])
        rev = cur.fetchall()[0]
        conn.commit()
        return Review(id=rev[0], rating=rev[1], song_name=rev[2], artist=rev[3])

    def delete_review(id):
        cur.execute("DELETE FROM review WHERE id = %s RETURNING *", [id])
        rev = cur.fetchall()[0]
        conn.commit()
        return Review(id=rev[0], rating=rev[1], song_name=rev[2], artist=rev[3])
    

